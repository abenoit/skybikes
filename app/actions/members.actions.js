import { SIGN_UP, LOGIN_SUCCESS, LOGOUT } from "../constants/actionTypes";

export const login = email => {
  return { type: LOGIN_SUCCESS, email };
};

export const logout = () => {
  return { type: LOGOUT };
};

export const signup = (firstname, lastname, email, phone) => {
  return { type: SIGN_UP, firstname, lastname, email, phone };
};
