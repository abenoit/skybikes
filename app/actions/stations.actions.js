import { RENT_BIKE, RETURN_BIKE } from "../constants/actionTypes";

export const rentBike = stationId => {
  return {
    type: RENT_BIKE,
    stationId
  };
};

export const returnBike = stationId => {
  return {
    type: RETURN_BIKE,
    stationId
  };
};
