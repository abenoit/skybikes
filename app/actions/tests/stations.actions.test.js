import { rentBike, returnBike, lateReturn } from "../stations.actions";

describe("Stations Actions", () => {
  const email = "test@test.com";
  it("rentBike", () => {
    const stationId = "1";
    expect(rentBike(stationId)).toMatchSnapshot();
  });

  it("returnBike", () => {
    const stationId = "1";
    expect(returnBike(stationId)).toMatchSnapshot();
  });

  it("lateReturn", () => {
    expect(lateReturn(email)).toMatchSnapshot();
  });
});
