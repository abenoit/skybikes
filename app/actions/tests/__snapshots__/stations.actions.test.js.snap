// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stations Actions lateReturn 1`] = `
Object {
  "member": "test@test.com",
  "type": "LATE_RETURN",
}
`;

exports[`Stations Actions rentBike 1`] = `
Object {
  "email": undefined,
  "stationId": "1",
  "type": "RENT_BIKE",
}
`;

exports[`Stations Actions returnBike 1`] = `
Object {
  "email": undefined,
  "stationId": "1",
  "type": "RETURN_BIKE",
}
`;
